src/VisualStudio/Core/Def/Implementation/ProjectSystem/Extensions/IVsHierarchyExtensions.cs

```diff
@@ -45,25 +45,25 @@ public static bool TryGetGuidProperty(this IVsHierarchy hierarchy, __VSHPROPID p
             => ErrorHandler.Succeeded(hierarchy.GetGuidProperty(VSConstants.VSITEMID_ROOT, (int)propertyId, out guid));
 
         public static bool TryGetProject(this IVsHierarchy hierarchy, [NotNullWhen(returnValue: true)] out EnvDTE.Project? project)
-            => hierarchy.TryGetProperty(__VSHPROPID.VSHPROPID_ExtObject, out project);
+            => hierarchy.TryGetProperty<EnvDTE.Project>(__VSHPROPID.VSHPROPID_ExtObject, out project);
```

Comment:
?

Reply:
It inferred `EnvDTE.Project?`, which is not what we want.